torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -318,7 +318,7 @@ PyObject* THPVariable_getitem(PyObject* self, PyObject* index) {
   variable_list variableIndices;
   int64_t specified_dims = count_specified_dimensions(holder.get());
   if (specified_dims == -1) {
-    return handle_torch_function_indexing(self, index);
+    return handle_torch_function_indexing(self, holder.get());
```

Comment:
Wasn't the point here that the user's `__torch_function__` would be called with the original argument and not the newly created tuple?

Also can you give more details how this was causing an error?

Reply:
So that function unpacks tuple but not list ? (I tried to look but it is a bit hard to follow)

Reply:
Normalization is OK, we do this occasionally, there is not an invariant that the arguments match up exactly. On the buggy case this logic

```
  if (PyTuple_Check(index)) {
    index_tup = py::reinterpret_borrow<py::object>(index);
  }
  else {
    index_tup = py::make_tuple(py::handle(index));
  }
```

causes index_tup to be `([T(), T()],)` but only the first tuple is iterated over. We could fix this but I'm happy to take this PR as is.